### PR TITLE
WIP resonsive images patch for core

### PR DESCRIPTION
**Pull request by joemcgill**
This PR is simply to make it easier to do code reviews before we have an initial patch, at which point we'll iterate directly in Trac.

See https://core.trac.wordpress.org/ticket/33641

### CODE REVIEW

File: src/wp-includes/media.php [hunk 1]
```diff
@@ -1173,7 +1303,7 @@ function wp_underscore_playlist_templates() {
 	<img src="{{ data.thumb.src }}"/>
 	<# } #>
 	<div class="wp-playlist-caption">
-		<span class="wp-playlist-item-meta wp-playlist-item-title"><?php
```
Comment (joemcgill) [inline src/wp-includes/media.php]: I need to revert this whitespace change.

File: tests/phpunit/includes/factory.php [hunk 2]
```diff
@@ -94,6 +94,35 @@ class WP_UnitTest_Factory_For_Attachment extends WP_UnitTest_Factory_For_Post {
 	function create_object( $file, $parent = 0, $args = array() ) {
 		return wp_insert_attachment( $args, $file, $parent );
 	}
+
+	function create_upload_object( $file, $parent = 0 ) {
```
Comment (joemcgill) [inline tests/phpunit/includes/factory.php:98]: Adding this factory method is something wonderboymusic and I discussed after https://core.trac.wordpress.org/changeset/33807, which could replace redundant `_make_attachment` helper functions  in `tests/image/intermediate_size.php`, `tests/post/attachment.php`, and `tests/ajax/MediaElement.php`. I still need to create a separate ticket for that.

File: src/wp-includes/media.php [hunk 3]
```diff
@@ -793,6 +810,119 @@ function wp_get_attachment_image($attachment_id, $size = 'thumbnail', $icon = fa
 }
 
 /**
+ * Get an array of image sources candidates for use in a 'srcset' attribute.
```
Comment (joemcgill) [inline src/wp-includes/media.php]: @todo: Improve doc description here to be less RICG jargony

**Comment by joemcgill** [comment]
Replace with #2

File: src/wp-includes/media.php [hunk 4]
```diff
@@ -793,6 +810,119 @@ function wp_get_attachment_image($attachment_id, $size = 'thumbnail', $icon = fa
...
+ *
+ * @since 4.4.0
+ *
+ * @param  int    $id   Image attachment ID.
+ * @param  string $size Optional. Name of image size. Default value: 'medium'.
+ * @return array|bool  An array of of srcset values or false.
```
Comment (getsource) [inline src/wp-includes/media.php:820]: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-1-parameters-that-are-arrays

File: src/wp-includes/media.php [hunk 5]
```diff
@@ -793,6 +810,119 @@ function wp_get_attachment_image($attachment_id, $size = 'thumbnail', $icon = fa
...
+	// See which image is being returned and bail if none is found.
+	if ( ! $img = wp_get_attachment_image_src( $id, $size ) ) {
+		return false;
+	}
+
+	// Break image data into url, width, and height.
+	list( $img_url, $img_width, $img_height ) = $img;
+
+	// Bail early If we have no width or if the width set to zero.
+	if ( ! $img_width > 0 ) {
```
Comment (getsource) [inline src/wp-includes/media.php:832]: Why does the width matter here, but not the height?

**Pull request closed without merging**